srecord/input/filter/byte_swap.cc

```diff
@@ -29,7 +29,7 @@ srecord::input_filter_byte_swap::~input_filter_byte_swap()
 srecord::input_filter_byte_swap::input_filter_byte_swap(
         const srecord::input::pointer &arg) :
     srecord::input_filter(arg),
-    buffer(),
+    
```

Comment:
I'm torn with this one. 
1. I understand that the default constructor would be called but explicitly laying out each member is nice and clear
2. We are moving default values to the declaration (see earlier check) and therefore only including non-default values in the constructor has merit...

I possibly just talked myself into this (lol). If you agree, let's make this check & changes

Reply:
I think was well covered elsewhere. I'm good with including this.